components/gitpod-cli/cmd/rebuild.go

```diff
@@ -0,0 +1,232 @@
+			return
+		}
+		defer client.Close()
+
+		wsInfo, err := client.Info.WorkspaceInfo(ctx, &api.WorkspaceInfoRequest{})
+		if err != nil {
+			utils.LogError(ctx, err, "Could not fetch the workspace info", client)
+			return
+		}
+
+		tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+		if err != nil {
+			log.Fatal("Could not create temporary directory")
+			return
+		}
+		defer os.RemoveAll(tmpDir)
+
+		event := utils.TrackEvent(ctx, client, &utils.TrackCommandUsageParams{
+			Command: cmd.Name(),
+		})
+		defer event.Send(ctx)
+
+		gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+		if err != nil {
+			fmt.Println("The .gitpod.yml file cannot be parsed: please check the file and try again")
+			fmt.Println("")
+			fmt.Println("For help check out the reference page:")
+			fmt.Println("https://www.gitpod.io/docs/references/gitpod-yml#gitpodyml")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if gitpodConfig == nil {
+			fmt.Println("To test the image build, you need to configure your project with a .gitpod.yml file")
+			fmt.Println("")
+			fmt.Println("For a quick start, try running:\n$ gp init -i")
+			fmt.Println("")
+			fmt.Println("Alternatively, check out the following docs for getting started configuring your project")
+			fmt.Println("https://www.gitpod.io/docs/configure#configure-gitpod")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MissingGitpodYaml)
+			return
+		}
+
+		var baseimage string
+		switch img := gitpodConfig.Image.(type) {
+		case nil:
+			baseimage = ""
+		case string:
+			baseimage = "FROM " + img
+		case map[interface{}]interface{}:
+			dockerfilePath := filepath.Join(wsInfo.CheckoutLocation, img["file"].(string))
+
+			if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+				fmt.Println("Your .gitpod.yml points to a Dockerfile that doesn't exist: " + dockerfilePath)
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileNotFound).Send(ctx)
+				return
+			}
+			dockerfile, err := os.ReadFile(dockerfilePath)
+			if err != nil {
+				log.Fatal("Could not read the Dockerfile")
+				return
+			}
+			if string(dockerfile) == "" {
+				fmt.Println("Your Gitpod's Dockerfile is empty")
+				fmt.Println("")
+				fmt.Println("To learn how to customize your workspace, check out the following docs:")
+				fmt.Println("https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-custom-dockerfile")
+				fmt.Println("")
+				fmt.Println("Once you configure your Dockerfile, re-run this command to validate your changes")
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileEmpty)
+				return
+			}
+			baseimage = "\n" + string(dockerfile) + "\n"
+		default:
+			fmt.Println("Check your .gitpod.yml and make sure the image property is configured correctly")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if baseimage == "" {
+			// TODO: should this fallback to workspace full?
```

Comment:
Should it be a special error code?

Reply:
Added `event.Set("ErrorCode", utils.RebuildErrorCode_NoCustomImage)`